runners/flink/src/test/java/org/apache/beam/runners/flink/translation/wrappers/streaming/DoFnOperatorTest.java

```diff
@@ -1038,7 +1039,7 @@ void pushbackDataCheckpointing(
 
     assertThat(
         stripStreamRecordFromWindowedValue(testHarness.getOutput()),
-        contains(helloElement, worldElement));
+        containsInAnyOrder(helloElement, worldElement));
```

Comment:
Why is the order reversed here?

Reply:
I think this is the last issue, will try to investigate today.

Reply:
We can't guarantee the order here for keys in the state backend after restoring from a savepoint. This just so happens to work for older Flink versions. So seems fair to change this.